### PR TITLE
Add service annotations

### PR DESCRIPTION
**Pull request by herbguo**
## Which problem is this PR solving?
When I use the jaeger operator, I need to add the annotations of the jaeger-query or jaeger-collector service. But the jaeger-operator do not supply the function. So I alter the code to supply this.

## Short description of the changes
-  I didn't alter the jaeger's type , just use the Annotation field. Before, it's just use to the component deployment, now I add the code to use it to the component service .
## For the issue
Closes https://github.com/jaegertracing/jaeger-operator/issues/1534
Closes https://github.com/jaegertracing/jaeger-operator/issues/1096

### CODE REVIEW

File: pkg/service/collector.go [hunk 1]
```diff
@@ -48,6 +48,7 @@ func collectorService(jaeger *v1.Jaeger, selector map[string]string) *corev1.Ser
 			Name:      GetNameForCollectorService(jaeger),
 			Namespace: jaeger.Namespace,
 			Labels:    util.Labels(GetNameForCollectorService(jaeger), "service-collector", *jaeger),
+			Annotations: jaeger.Spec.Collector.Annotations,
```
Comment (herbguo) [inline pkg/service/collector.go:51]: Add the Collector service Annotations

File: pkg/service/collector_test.go [hunk 2]
```diff
@@ -143,3 +143,13 @@ func TestCollectorServiceLoadBalancer(t *testing.T) {
 	// Only the non-headless service will receive the type
 	assert.Equal(t, svc[1].Spec.Type, corev1.ServiceTypeLoadBalancer)
 }
+func TestCollectorServiceAnnotations(t *testing.T) {
```
Comment (herbguo) [inline pkg/service/collector_test.go:146]: The test case of collector

File: pkg/service/query.go [hunk 3]
```diff
@@ -14,6 +14,9 @@ func NewQueryService(jaeger *v1.Jaeger, selector map[string]string) *corev1.Serv
 	trueVar := true
 
 	annotations := map[string]string{}
+	if jaeger.Spec.Query.Annotations !=nil{
```
Comment (herbguo) [inline pkg/service/query.go:17]: Add the Query service Annotations

File: pkg/service/query_test.go [hunk 4]
```diff
@@ -109,3 +109,20 @@ func TestQueryServiceSpecifiedNodePortWithIngress(t *testing.T) {
 	assert.Equal(t, intstr.FromInt(16686), svc.Spec.Ports[0].TargetPort)
 	assert.Equal(t, svc.Spec.Type, corev1.ServiceTypeNodePort)
 }
+
```
Comment (herbguo) [inline pkg/service/query_test.go:112]: The test case of query

**Review by herbguo — COMMENTED** [review]
I didn't alter the jaeger's type , just use the Annotation field. Before, it's just use to the component deployment, now I add the code to use it to the component service .

**Comment by pavolloffay** [comment]
@rubenvp8510 could you please take a look and review the PR.

**Review by frzifus — COMMENTED** [review]
Is that still a thing? if yes, @herbguo could you try to `go fmt` your changes? Seems they are not formated.

**Comment by herbguo** [comment]
> Ahh the [ci thing](https://github.com/jaegertracing/jaeger-operator/runs/5630239008?check_suite_focus=true#step:6:100) was also failing. calling `make format` should fix it.

I have performed the fix operation, waiting for the review.

**Review by frzifus — APPROVED** [review]
LGTM, @rubenvp8510 can we go ahead?

**Review by rubenvp8510 — APPROVED** [review]

**Pull request merged**